apps/meteor/tests/e2e/delete-account.spec.ts

```diff
@@ -0,0 +1,133 @@
+import { DEFAULT_USER_CREDENTIALS } from './config/constants';
+import { AccountProfile, Registration, Utils } from './page-objects';
+import { ToastBar } from './page-objects/toastBar';
+import { test, expect } from './utils/test';
+import { createTestUser, type ITestUser } from './utils/user-helpers';
+
+test.describe('Delete Own Account', () => {
+	let poAccountProfile: AccountProfile;
+	let poRegistration: Registration;
+	let poToastBar: ToastBar;
+	let poUtils: Utils;
+	let userToDelete: ITestUser;
+	let userWithInvalidPassword: ITestUser;
+	let userWithoutPermissions: ITestUser;
+
+	test.beforeAll(async ({ api }) => {
+		const response = await api.post('/settings/Accounts_AllowDeleteOwnAccount', { value: true });
+		expect(response.status()).toBe(200);
+		userToDelete = await createTestUser(api, { username: 'user-to-delete' });
+		userWithInvalidPassword = await createTestUser(api, { username: 'user-with-invalid-password' });
+		userWithoutPermissions = await createTestUser(api, { username: 'user-without-permissions' });
+	});
+
+	test.beforeEach(async ({ page }) => {
+		poAccountProfile = new AccountProfile(page);
+		poRegistration = new Registration(page);
+		poToastBar = new ToastBar(page);
+		poUtils = new Utils(page);
+		await page.goto('/home');
+	});
+
+	test.afterAll(async ({ api }) => {
+		const response = await api.post('/settings/Accounts_AllowDeleteOwnAccount', { value: false });
+		expect(response.status()).toBe(200);
+		await userToDelete.delete();
+		await userWithInvalidPassword.delete();
+		await userWithoutPermissions.delete();
```

Comment:
Could you add these calls to a `Promise.all`? 
This will help prevent situations where the other calls don't execute because one of them failed.